hyperspy/hyperspy_extension.yaml

```diff
@@ -0,0 +1,229 @@
+    signal_dimension: 1
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.signal1d
+  Signal2D:
+    signal_type: ""
+    signal_dimension: 2
+    dtype: real
+    lazy: False
+    module: hyperspy._signals.signal2d
+  LazySignal2D:
+    signal_type: ""
+    signal_dimension: 2
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.signal2d
+  ComplexSignal:
+    signal_type: ""
+    signal_dimension: -1
+    dtype: complex
+    lazy: False
+    module: hyperspy._signals.complex_signal
+  LazyComplexSignal:
+    signal_type: ""
+    signal_dimension: -1
+    dtype: complex
+    lazy: True
+    module: hyperspy._signals.complex_signal
+  ComplexSignal1D:
+    signal_type: ""
+    signal_dimension: 1
+    dtype: complex
+    lazy: False
+    module: hyperspy._signals.complex_signal1d
+  LazyComplexSignal1D:
+    signal_type: ""
+    signal_dimension: 1
+    dtype: complex
+    lazy: True
+    module: hyperspy._signals.complex_signal1d
+  ComplexSignal2D:
+    signal_type: ""
+    signal_dimension: 2
+    dtype: complex
+    lazy: False
+    module: hyperspy._signals.complex_signal2d
+  LazyComplexSignal2D:
+    signal_type: ""
+    signal_dimension: 2
+    dtype: complex
+    lazy: True
+    module: hyperspy._signals.complex_signal2d
+  HologramImage:
+    signal_type: hologram
+    signal_dimension: 2
+    dtype: real
+    lazy: False
+    module: hyperspy._signals.hologram_image
+  LazyHologramImage:
+    signal_type: hologram
+    signal_dimension: 2
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.hologram_image
+  EELSSpectrum:
+    signal_type: EELS
+    signal_type_aliases:
+      - TEM EELS
+    signal_dimension: 1
+    dtype: real
+    lazy: False
+    module: hyperspy._signals.eels
+  LazyEELSSpectrum:
+    signal_type: EELS
+    signal_type_aliases:
+      - TEM EELS
+    signal_dimension: 1
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.eels
+  EDSTEMSpectrum:
+    signal_type: EDS_TEM
+    signal_dimension: 1
+    dtype: real
+    lazy: False
+    module: hyperspy._signals.eds_tem
+  LazyEDSTEMSpectrum:
+    signal_type: EDS_TEM
+    signal_dimension: 1
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.eds_tem
+  EDSSEMSpectrum:
+    signal_type: EDS_SEM
+    signal_dimension: 1
+    dtype: real
+    lazy: False
+    module: hyperspy._signals.eds_sem
+  LazyEDSSEMSpectrum:
+    signal_type: EDS_SEM
+    signal_dimension: 1
+    dtype: real
+    lazy: True
+    module: hyperspy._signals.eds_sem
+  DielectricFunction:
+    signal_type: DielectricFunction 
+    signal_type_aliases:
+      - dielectric function
+    signal_dimension: 1
+    dtype: complex
+    lazy: False
+    module: hyperspy._signals.dielectric_function
+  LazyDielectricFunction:
+    signal_type: DielectricFunction
+    signal_type_aliases:
+      - dielectric function
+    signal_dimension: 1
+    dtype: complex
+    lazy: True
+    module: hyperspy._signals.dielectric_function
+
+components1D:
```

Comment:
Good point. In principle that could be done, as we can filter the components later on. However, filtering them at the yaml level i) saves some code ii) makes more explicit what sort of component the extension is adding.

I am happy with both, so I let you decide. Do let me know.

Reply:
I think I would prefer to have:
```yml
signals:
  MySignal:
    signal_type: "MySignal"
    signal_dimension: 1
    dtype: real
    lazy: False
    module: my_package.signal
components:
  MyComponentClass:
    signal_dimension: 1
    module: my_package.components
```
Because components and signals would at the same level and it would be "symmetrical".
We could also consider changing the API and have all the HyperSpy components in `hs.components`, but this is a discussions for another time! ;)